scripts/sync-down-from-production.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env sh
+set -xe
+
+SCRIPT_DIR=$(dirname "$0")
+
+# Load up environment variables for the local (docker) postgres
+. "$SCRIPT_DIR"/../.env
+
+# Check wheter `docker compose` or `docker-compose` is available
+if command -v docker-compose > /dev/null; then
+  DOCKER_COMPOSE=docker-compose
+elif command -v docker compose > /dev/null; then
+  DOCKER_COMPOSE="docker compose"
+else
+  echo "Neither docker compose nor docker-compose is available"
+  exit 1
+fi
+
+REMOTE_PG="${PRODUCTION_PG_URL_FOR_USER_GITHUB_ACTIONS}"
+LOCAL_PG="postgres://${PG_USERNAME}:${PG_PASSWORD}@postgres/${PG_DATABASE}"
+
+# run postgres utils inside a container so that we don't have to install them on the host
+pg_exec() {
+  (cd "${SCRIPT_DIR}/.." && ${DOCKER_COMPOSE} exec -T postgres bash -c "${*}")
+}
+
+# truncate tables users and teams and cascade
+pg_exec "psql --command='TRUNCATE TABLE users, teams CASCADE;' ${LOCAL_PG}"
+# insert users from production
+pg_exec "psql --command=\"\\COPY (SELECT * FROM users) TO STDOUT (FORMAT CSV, DELIMITER ';')\" \"${REMOTE_PG}\" | psql --command=\"\\COPY users FROM STDIN (FORMAT CSV, DELIMITER ';')\" \"${LOCAL_PG}\""
+# insert teams from production
+pg_exec "psql --command=\"\\COPY (SELECT * FROM teams) TO STDOUT (FORMAT CSV, DELIMITER ';')\" \"${REMOTE_PG}\" | psql --command=\"\\COPY teams FROM STDIN (FORMAT CSV, DELIMITER ';')\" \"${LOCAL_PG}\""
+# insert flows from production and set version to 1
+pg_exec "psql --command=\"\\COPY (SELECT * FROM flows) TO STDOUT (FORMAT CSV, DELIMITER ';')\" \"${REMOTE_PG}\" | psql --command=\"\\COPY flows FROM STDIN (FORMAT CSV, DELIMITER ';'); UPDATE flows SET version = 1;\" \"${LOCAL_PG}\""
+# insert an operation for each flow (to make sharedb happy)
+SQL=$(cat <<EOF
+INSERT INTO operations (flow_id, data, version) 
+  SELECT
+    id
+  , json_build_object(
+      'm', json_build_object(
+        'ts', extract(epoch from now()) * 1000
+      , 'uId', '1'
+      )
+      , 'v', 0
+      , 'seq', 1
+      , 'src', '1'
+      , 'create', json_build_object(
+        'data', '{}'
+        , 'type', 'http://sharejs.org/types/JSONv0'
+      )
+    )
+  , 1
+  FROM flows;
+EOF
+)
+pg_exec "psql --command=\"${SQL}\" \"${LOCAL_PG}\""
+# insert published_flows (the last two)
+SQL="\\COPY (SELECT id, data, flow_id, summary, publisher_id FROM (SELECT id, data, flow_id, summary, publisher_id, ROW_NUMBER() OVER (PARTITION BY flow_id ORDER BY created_at DESC) as row_num FROM published_flows) as subquery WHERE row_num <= 2) TO STDOUT (FORMAT CSV);"
+pg_exec "psql --command=\"${SQL}\" \"${REMOTE_PG}\" | psql --command=\"\\COPY published_flows (id, data, flow_id, summary, publisher_id) FROM STDIN (FORMAT CSV)\" \"${LOCAL_PG}\""
+# (now with `users` sync, this might not be needed anymore) set published_flows.publisher_id=1 to prevent errors with non-existing ID constraint
+
+
+
+# TODO: Do all of this in a transaction ideally
```

Comment:
:heavy_plus_sign: would second wrapping in a transaction so we have a safe rollback on failure / don't end up in with a fully truncated but partially synced db etc